### PR TITLE
bugfix:fix seata-spring-boot-starter does not work

### LINKED ISSUE #2027: 1.0版使用yml配置时启动异常
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
引入seata-all或者seata-spring-boot-starter时,按照script文件夹内的yml配置时启动异常

### Ⅱ. Describe what happened

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'globalTransactionScanner' defined in class path resource [io/seata/spring/boot/autoconfigure/SeataAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.seata.spring.annotation.GlobalTransactionScanner]: Factory method 'globalTransactionScanner' threw exception; nested exception is java.lang.ExceptionInInitializerError
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:627) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:456) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1321) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1160) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:555) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:515) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:320) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:204) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.registerBeanPostProcessors(PostProcessorRegistrationDelegate.java:228) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.registerBeanPostProcessors(AbstractApplicationContext.java:721) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:534) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141) ~[spring-boot-2.1.8.RELEASE.jar:2.1.8.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:744) [spring-boot-2.1.8.RELEASE.jar:2.1.8.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:391) [spring-boot-2.1.8.RELEASE.jar:2.1.8.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:312) [spring-boot-2.1.8.RELEASE.jar:2.1.8.RELEASE]
	at org.test.ProviderApplication.main(ProviderApplication.java:22) [classes/:?]
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.seata.spring.annotation.GlobalTransactionScanner]: Factory method 'globalTransactionScanner' threw exception; nested exception is java.lang.ExceptionInInitializerError
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:622) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	... 17 more
Caused by: java.lang.ExceptionInInitializerError
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:90) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:133) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:122) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration.globalTransactionScanner(SeataAutoConfiguration.java:57) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a.CGLIB$globalTransactionScanner$1(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a$$FastClassBySpringCGLIB$$d8c42c73.invoke(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:363) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a.globalTransactionScanner(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_201]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:622) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	... 17 more
Caused by: java.lang.NullPointerException
	at io.seata.config.FileConfiguration.<init>(FileConfiguration.java:96) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.config.ConfigurationFactory.<clinit>(ConfigurationFactory.java:58) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:90) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:133) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:122) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration.globalTransactionScanner(SeataAutoConfiguration.java:57) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a.CGLIB$globalTransactionScanner$1(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a$$FastClassBySpringCGLIB$$d8c42c73.invoke(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:363) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a.globalTransactionScanner(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_201]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:622) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	... 17 more

```
### Ⅳ. How to reproduce it (as minimally and precisely as possible)

引入按yml配置即可重现

- JDK version :1.8
- OS :windows
- Others:

### PR DESCRIPTION
**Pull request by xingfudeshi**
### Ⅰ. Describe what this PR did
fix seata-spring-boot-starter does not work.

### Ⅱ. Does this pull request fix one issue?

fixes #2027

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
I've done integration testing.

### CODE REVIEW

File: config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java [hunk 1]
```diff
@@ -92,13 +103,22 @@ public FileConfiguration(String name, boolean allowDynamicRefresh) {
...
-            fileConfig = ConfigFactory.load(name);
+            //conf files may not exist when using seata-spring-boot-starter
+            URL resource = this.getClass().getClassLoader().getResource(name);
+            if (null != resource) {
+                targetFilePath = resource.getPath();
+                fileConfig = ConfigFactory.load(name);
+                targetFileLastModified = new File(targetFilePath).lastModified();
+                this.allowDynamicRefresh = allowDynamicRefresh;
+            } else {
+                targetFilePath = null;
```
Comment (slievrly) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: defensive code.

**Comment by zjinlei** [comment]
retry ci

File: config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java [hunk 2]
```diff
@@ -92,13 +103,22 @@ public FileConfiguration(String name, boolean allowDynamicRefresh) {
             targetFilePath = name.substring(SYS_FILE_RESOURCE_PREFIX.length());
```
Comment (zjinlei) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: targetFileLastModified is null
tempLastModified > targetFileLastModified will be NPE

Comment (xingfudeshi) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: When allowDynamicRefresh is  false,the code won't be executed.

Comment (zjinlei) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: When config.type is file and file.name is "file: ..."

Comment (xingfudeshi) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: OK.

Comment (zjinlei) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: Here and ConfigurationFactory have been iterated many times, it is difficult to read now, hope can be refactored.

Comment (xingfudeshi) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: NPE fixed

**Pull request merged**